### ISSUE TITLE
Introduce a Points System

### ISSUE THREAD

**Issue by harrygfox** [cul-2016/quiz#504]
@stianr 
# Why

@nogainbar and I feel that their are a number issues with the app designs. 
- There is not enough in-app motivation for the students to up their participation levels
- the leaderboard being ordered by a score (which we have determined through trial and error) practically undermines the value of medals. 
- the module leaderboard over-emphasises the value of correct answers rather than, what we believe to be the true purpose of the app, learning. 

# What

We suggest a points system to apply more measured weighting and value to each achievable element. For example a point system could go as follows:

| Achievable Element | Points Value |
| ------- | ------- |
| Your Percentage Score in a Quiz | _[Your percentage]_ * 100 |
| Gold Medal | 100 |
| Silver Medal | 50 |
| Bronze Medal | 25 |
| '1st Quiz' trophy | 50 |
| 'Participation' trophy | 150 |
| '100% Score' trophy | 100 |
| 'Beaten the Overall Average Score' trophy | 100 |

Furthermore, with additional layers to trophies there could be different numbers of points to be won.  E.g. Participation in 5 quizes is worth `200` but participation in 10 quizes is worth `500`

## Implications for the Leaderboard

This scoring would replace the current method for ordering the leaderboard - instead of ordering by highest number of correct answers, students would be ordered by points (or quodl-coins or whatever we wanted to call them). The fastest way to climb the leaderboard is now, not only to get correct answers, but to get gold medals and, importantly, turn up to lectures. 

## Examples:
It is my first quiz in this module.
On a quiz with 15 questions I score 11 correct answers.
My percentage score is 73%, just enough to earn a gold medal.
At the end of the quiz I see:

_Congratulations you scored `11/15`_ ... `+73`
_You have been awarded a Gold Medal!_ `+100`
_Trophy unlocked! "My 1st Quiz"_ `+100`
_Your Score: 273 | Rank for this quiz: 3rd_

I look up at the lecture hall screen to see the lecturer with the Module leaderboard showing me that I have moved to 3rd position for the module and I get a sense of the other scores gained.

----
What are your thoughts?

**Comment by stianr** [comment]
This is an interesting idea. My thoughts are around:

- Lecturer and student buy-in. A points-based leaderboard is transparent to everyone, and needs no introduction or explanation. Think of a busy lecturer, who wants to make their lectures engaging, but has ten minutes to spend getting to grips with Quodl. The scoring system will catch them for a while, and many will have the response "Why can't we just have a leaderboard". Students will approach them at the end of a lecture saying "Why do I have a lower score than my friend when I got more questions right", and the lecturer won't be able to answer. The lecturer will be annoyed and the student will think it's unfair. So I feel strongly that the primary way of presenting the leaderboard should be on overall score. 

- That said, having separate scores for performance (just the number of questions correctly answered) and engagement (including things like number of quizzes attempted, number of reviews of previous quizzes etc.) could work. Lecturers could choose to ignore the engagement score if they wanted, but it would be there for those who took the time to make sense of what it was. This would require (a) an extra column on the leaderboard, (b) queries to retrieve relevant data, (c) algorithm for calculating score, (d) ability to order leaderboard by either performance or engagement score.

- If we revert to the leaderboard showing the status of the trophies/badge/whatever we call them, next to each user on the leaderboard, this allows lecturers to reward participation as well as performance. (What I did with the prototype last year was to go through the leaderboard and give a prize to everyone who managed to get all four trophies, as well as prizes for the top performers.) This leads to a similar outcome as using a points-based currency, but with minimal development requirements.

- Given the development time we have available I don't think introducing a more complex scoring system provides enough of a benefit relative to the other things we'd like to do.
